feat: Add support for markdown descriptions for models, in the Explore sidebar models list

filp opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

Follow-up to #1174 
Ticket: #9148 

### Description:

Extends markdown/longer description support to the Explore sidebar's models list

![Kapture 2024-02-29 at 10 42 55](https://github.com/lightdash/lightdash/assets/382538/6437f3ac-6478-4f5d-990b-96d5ae13c842)
![Kapture 2024-02-29 at 13 44 54](https://github.com/lightdash/lightdash/assets/382538/ed50b0a2-778e-41bc-8e07-8a08c1cf6b6d)

- Removes tooltip from breadcrumb for simplicity for now, will re-introduce it with markdown support on a follow-up if deemed necessary, but requires more work to ensure it works well across different breadcrumb uses.
- Adds support for markdown descriptions on models
- Reshuffles some code around to remove duplication around model descriptions + minor cleanup


### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

hamzahc1 commented:
@filp do we want to open a separate issue for the related explores so we can track it?

filp commented on packages/frontend/src/components/Explorer/ExploreSideBar/index.tsx:
```diff
@@ -180,15 +181,17 @@ const ExploreSideBar = memo(() => {
-                )}
-            </Stack>
-        </TrackSection>
+        <ItemDetailProvider>
```
Pulled the provider higher up in the tree to also support model descriptions.

filp commented on packages/frontend/src/components/Explorer/ExploreTree/TableTree/ItemDetailPreview.tsx:
```diff
@@ -0,0 +1,89 @@
+import { Anchor, Box, Flex, Title, useMantineTheme } from '@mantine/core';
```
Extracted these components out into their own file.

filp commented:
@hamzahc1 No longer necessary, added the missing piece directly as part of this PR, just required a bit of shuffling code around.

IrakliJani commented on packages/frontend/src/components/Explorer/ExploreTree/TableTree/ItemDetailPreview.tsx:
```diff
@@ -0,0 +1,89 @@
+                    pt={20}
+                    style={{
+                        background:
+                            'linear-gradient(rgba(255,255,255,0) 0%, rgba(255,255,255,1) 60%, rgba(255,255,255,1) 100%)',
```
there are some mantine helpers functions to create gradients from mantine colors. although im ok with this css since its just a white color / transparent color. btw I think you can use `transparent` instead of `rgba(255,255,255,1)`.

IrakliJani commented on packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeSingleNode.tsx:
```diff
@@ -11,121 +11,30 @@ import {
-                     * a slight fade effect.
-                     */
-                    mt={-30}
-                    pt={20}
```
do we need this?

IrakliJani commented on packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeSingleNode.tsx:
```diff
@@ -11,121 +11,30 @@ import {
-                     * Forces the 'Read more' option to slightly overlap with the content, and show
-                     * a slight fade effect.
-                     */
-                    mt={-30}
```
To avoid hardcoded values and negative margin-top, we could change this div to absolute positioning and make the popover relative. Then, you can simply use bottom={0}

IrakliJani reviewed (commented):
this looks great!

something I noticed:

hovercard seems detached from the "trigger." Maybe adding an arrow will help maintain the connection.
https://github.com/lightdash/lightdash/assets/962095/29c400b4-2c6e-43a0-aef4-3b2ec6a4abc8

filp commented:
Thanks @IrakliJani - made some tweaks with your suggestions in consideration:

![Kapture 2024-03-01 at 01 44 29](https://github.com/lightdash/lightdash/assets/382538/45a22e18-741c-4ffe-8c86-e29486d81634)

Basically:

- Added arrows to the popovers
- Changed the preview fade out effect to not require any sort of absolute positioning or hard-coded negative margins. The effect is instead achieved with a CSS mask image, which has pretty good browser support (and the fallback behavior is absolutely fine).
- Tweaked offsets to prevent the popover from overlapping with the model/table chevron icon.

IrakliJani commented on packages/frontend/src/components/Explorer/ExploreTree/TableTree/ItemDetailPreview.tsx:
```diff
@@ -0,0 +1,159 @@
+                    overflow: 'hidden',
+
+                    // If we're over the truncation limit, use a mask to fade out the bottom of the container.
+                    maskImage: isTruncated
+                        ? 'linear-gradient(180deg, white 0%, white 80%, transparent 100%)'
+                        : undefined,
```
oh nice! I've never used these

filp replied:
Yeah same! I had the idea in the back of my head it was a viable solution, but had never actually used it.

Forgot to mention re: using a mantine helper for gradients: ended up not using them because they are not very flexible in terms of how the gradient steps are spaced out!

IrakliJani replied:
Agreed! It can be useful only if you're using Mantine colors and you don't want to hardcode a HEX. Otherwise, this approach is more readable.

ZeRego commented:
@all-contributors add @filp for code.
